src/Data/Functor/Loom.hs

```diff
@@ -3,90 +3,110 @@
+runLoom (Loom ctx eta) m = eta (m <$ ctx)
 {-# INLINE runLoom #-}
 
--- | Composition of weaving functions with 'Loom'.
+-- | 'Loom' composition, 'Loom's can be constructed with the 'bind', 'lift', 'run', and 'hoist' combinators
+-- and subsequently composed to produce complex weaves in a straightforward way:
 --
--- @since 0.1.0.0
-weave ::
-  (Functor f, Functor o) =>
-  f () ->
-  (forall x. f (n x) -> o (f x)) ->
-  Loom m n a b ->
-  Loom m o a (f b)
-weave ctx eta (Loom ctx' eta' tmorphism) =
-  Loom (Compose (ctx' <$ ctx)) (Handler.compose eta' (Handler eta)) (fmap tmorphism . getCompose)
-{-# INLINE weave #-}
-
--- | Composition of 'Loom'.
+-- @
+-- -- weaving a state
+-- let loom :: Loom (Lang ctx effs') (Lang ctx effs) a b
+-- let weaveF :: s -> Lang ctx (eff ': effs) a -> Lang ctx effs (s, a)
+--     weaveF = uncurry (loom ~>~ weave (st :: s, ()) (f :: s -> a -> Lang ctx effs (s a)))
+-- @
 --
 -- @since 0.1.0.0
 infixr 9 ~>~
 
-(~>~) :: Functor o => Loom m n a b -> Loom n o b c -> Loom m o a c
-Loom ctx eta tmorph ~>~ Loom ctx' eta' tmorph' =
-  Loom (Compose (ctx <$ ctx')) (Handler.compose eta eta') (tmorph' . fmap tmorph . getCompose)
+(~>~) :: Loom m n a b -> Loom n o b c -> Loom m o a c
+Loom ctx eta ~>~ Loom ctx' eta' = Loom (Compose (ctx <$ ctx')) (eta' . fmap eta . getCompose)
 {-# INLINE (~>~) #-}
 
 -- | Constructs the identity 'Loom'.
 --
 -- @
--- id == Loom (Identity ()) (fmap Identity . runIdentity) runIdentity)
+-- id == Loom (Identity ()) runIdentity
 -- @
 --
 -- @since 0.1.0.0
-identity :: Functor m => Loom m m a a
-identity = Loom (Identity ()) Handler.identity runIdentity
+identity :: Loom m m a a
+identity = Loom (Identity ()) runIdentity
 {-# INLINE identity #-}
 
+-- | Constructs an "Effect Handlers in Scope"-style weaving function from the functor context and the distribution
+-- function. This is a synonym for the 'Loom' constructor.
+--
+-- @since 0.1.0.0
+weave :: Functor f => f () -> (f (m a) -> n (f b)) -> Loom m n a (f b)
+weave = Loom
+{-# INLINE weave #-}
+
+-- | Constructs a 'Loom' from a bind function.
+--
+-- @since 0.1.0.0
+bind :: Monad n => (a -> n b) -> Loom n n a b
+bind k = Loom (Identity ()) (runIdentity >=> k)
+{-# INLINE bind #-}
+
+-- | Lifts a function to a 'Loom'. This is equivalent to:
+--
+-- @
+-- 'hoist' ('fmap' f)
+-- @
+--
+-- @since 0.1.0.0
+lift :: Functor m => (a -> b) -> Loom m m a b
+lift f = Loom (Identity ()) (fmap f . runIdentity)
+{-# INLINE lift #-}
+
+-- | Constructs a 'Loom' from an function with a functorial result.
+--
+-- @since 0.1.0.0
+run :: (m a -> n (f a)) -> Loom m n a (f a)
+run runner = Loom (Identity ()) (runner . runIdentity)
+{-# INLINE run #-}
+
 -- | Constructs a 'Loom' from a natural transformation.
 --
 -- @since 0.1.0.0
-hoist :: Functor g => (f ~> g) -> Loom f g a a
-hoist eta = Loom (Identity ()) (Handler.hoist eta) runIdentity
+hoist :: (f a -> g b) -> Loom f g a b
+hoist eta = Loom (Identity ()) (eta . runIdentity)
```

Comment:
It looks like `run` is just a more specialized version of `hoist`? I'm curious why you have both.

Reply:
Thats true! I must have just missed the redundancy.